airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/write/PerformanceTestScenarios.kt

```diff
@@ -0,0 +1,188 @@
+            generationId = 0,
+            minimumGenerationId = 0,
+            syncId = 1,
+        )
+    }
+
+    private val random = SecureRandom()
+    private val randomThreshold: Int =
+        if (duplicateChance > 0.0) ((duplicateChance % 1.0) * 100).toInt() else 0
+
+    private var recordCount: Long = 0
+    private var byteCount: Long = 0
+
+    override val catalog = DestinationCatalog(listOf(stream))
+
+    class RecordWriter(
+        indexColumn: NamedField,
+        columns: List<NamedField>,
+        stream: DestinationStream,
+        private val destination: DestinationProcess,
+        private val recordBufferSize: Long = 1,
+    ) : AutoCloseable {
+        private val baseRecord = run {
+            val data = (listOf(indexColumn) + columns).associate { Pair(it.name, it.sample) }
+            InputRecord(
+                namespace = stream.descriptor.namespace,
+                name = stream.descriptor.name,
+                data = Jsons.serialize(data),
+                emittedAtMs = System.currentTimeMillis(),
+            )
+        }
+        private val messageParts =
+            Jsons.serialize(baseRecord.asProtocolMessage()).split(indexColumn.sample.toString())
+        private val baseMessageSize = messageParts.sumOf { it.length }
+
+        private val sb = StringBuilder()
+
+        var recordWritten: Long = 0
+        var bytesWritten: Long = 0
+
+        fun write(id: Long) {
+            sb.append(messageParts[0])
+            sb.append(id)
+            sb.append(messageParts[1])
+            sb.appendLine()
+
+            if (recordWritten % recordBufferSize == 0L) {
+                flush()
+            }
+
+            recordWritten += 1
+            bytesWritten += baseMessageSize + id.length()
+        }
+
+        private fun flush() {
+            if (sb.isNotEmpty()) {
+                destination.sendMessage(sb.toString())
+                sb.clear()
+            }
+        }
+
+        override fun close() {
+            flush()
+        }
+    }
+
+    override fun send(destination: DestinationProcess) {
+        RecordWriter(
+                indexColumn = idColumn,
+                columns = columns,
+                stream = stream,
+                destination = destination,
+                recordBufferSize = 10,
+            )
+            .use { writer ->
+                (1..recordsToInsert).forEach {
+                    writer.write(it)
+                    if (randomThreshold > 0 && random.nextInt(0, 100) <= randomThreshold) {
+                        writer.write(it)
+                    }
+                }
+                recordCount = writer.recordWritten
+                byteCount = writer.bytesWritten
+            }
+    }
+
+    override fun getSummary() =
+        PerformanceTestScenario.Summary(
+            recordCount,
+            byteCount,
+            expectedRecordsCount = if (dedup) recordsToInsert else recordCount,
+        )
+}
+
+private fun Long.length(): Long =
```

Comment:
Why not a toString here? I this performance critical?

Reply:
This is called in the record gen loop so I was trying to minimize it as much as possible. Impact vary with the complexity of the payload, it will matter more in simple payloads that bigger ones.